### ISSUE TITLE
Ethereal Soul Trader

### ISSUE THREAD

**Issue by Tamagucci** [WoWManiaUK/Redemption#2506]
Hi there. I've noticed an issue with the ethereal soul trader pet.

http://www.wowhead.com/spell=49964/ethereal-soul-trader

I talked to a GM and mentioned the issue and was directed here. Pretty much, the pet is suppose to collect an item called "ethereal credits" when you kill a mob and you can use those to purchase items the pet sells. I tested it on the Fallen Hero's Spirit in Ice Crown (level 80 mobs) that should have given the credits but the pet failed to collect (yes, I had him summoned) them as stated. 

http://www.wowhead.com/item=38186/ethereal-credit#created-by-spell

Thanks for looking into the issue! I was really excited to get this pet, it was my favorite in BC and really wanted to get the credits so I could buy the xmog gear from him.

Oh and p.s. - He doesnt do his emote when summoned as well. From wowhead "When summoned emotes "I have arrived. Shall we set to work, then?"." Not a big deal at all but would be nice to have fixed! 

Thanks!

**Comment by TheManiax** [comment]
**How it should work:**
To obtain a credit, you must first summon your [[Ethereal Soul-Trader]](www.wow-mania.com/armory/?spell=49964) which is done by using the [[Soul-Trader Beacon]](http://www.wow-mania.com/armory/?item=38050).
Once the Soul-Trader is summoned, you must kill players or NPCs which are no less than nine-levels below you.
This way you will obtain a [[Ethereal Credit]](http://www.wow-mania.com/armory/?item=38186) for each kill. Simultaneous kills might be counted only once. 
I couldn't find much info on web, it used to bug in blizzard often as well not getting credits (in most cases resummoning the pet used to solve the issue) , this pet was also changed a bit on later patches (5.0.5 & 5.4.8) by blizzard due to some exploits:
[Hotfix](https://worldofwarcraft.com/en-us/news/14216440) (2014-06-03): "Ethereal Soul-Trader can no longer be summoned while in an Arena or Battleground."
[Hotfix](https://worldofwarcraft.com/en-us/news/7207170) (2012-09-12): "The Ethereal Soul-Trader pet is now immune to player abilities."

**Comment by jmazzz** [comment]
Hey - thanks for the reply. I did some research too and apparently it has to do with the aura the pet applies onto the player -

"Looking on spellwork, this aura has correct proc flags (should proc on killing enemy who yield honor or xp), but no proc spell defined. We need to add to unit.cpp in the spell_generic case for handledummyauraproc"

This poster had the same issue and was apparently fixed over in trinity but I'm not a coder (far from it) so I don't know if its the same issue but they're having the same problem.

Here it is:

https://github.com/TrinityCore/TrinityCore/issues/4152

**Comment by Keader** [comment]
New Script implemented.
Ready to test in next restart.

**Comment by WhitephoenixTheBunny** [comment]
The pet only give credits on the 1st kill.

**Comment by Keader** [comment]
Fixes already.
Janitor Will deploy on Red
